test/integration/toolchains/cmake/test_cmaketoolchain.py

```diff
@@ -1046,6 +1046,30 @@ def test_set_cmake_lang_compilers_and_launchers():
     assert 'set(CMAKE_RC_COMPILER "C:/local/rc.exe")' in toolchain
 
 
+def test_cmake_presets_compiler():
+    profile = textwrap.dedent(r"""
+    [settings]
+    os=Windows
+    arch=x86_64
+    compiler=clang
```

Comment:
I can see the output is as expected, but wouldn't he compiler executables be `clang` or `clang-cl.exe` instead?

Reply:
I think this is being directly overriden by ``tools.build:compiler_executables`` and most likely the intention of the test was to have ``compiler=msvc``, can you please confirm @petwu?

I'll check if it makes sense or is possible to introduce some check for this.

Reply:
I suspect this is derived from copy&paste the test above called ``test_set_cmake_lang_compilers_and_launchers``

Reply:
You are right, I just copied the test above. Setting `compiler=clang` and then saying your executable is called `cl.exe` of course doesn't make sense in a real-world scenario. The test should just verify that the `compiler_executable` setting ended up in the generated preset.